### ISSUE TITLE
TestMockDecoration.test_wrapped_getfuncargnames_patching test fails with Python 2.7

### ISSUE THREAD

**Issue by marv** [pytest-dev/pytest#2294]
`python2.7 setup.py config`
`python2.7 setup.py build`
`python2.7 pytest.py testing`

results in 1 test failure:

```
> ======================================================================================= FAILURES =======================================================================================
> _______________________________________________________________ TestMockDecoration.test_wrapped_getfuncargnames_patching _______________________________________________________________
> 
> self = <integration.TestMockDecoration instance at 0x7f00bf92b488>
> 
>     def test_wrapped_getfuncargnames_patching(self):
>         from _pytest.compat import getfuncargnames
>     
>         def wrap(f):
>             def func():
>                 pass
>             func.__wrapped__ = f
>             func.patchings = ["qwe"]
>             return func
>     
>         @wrap
>         def f(x, y, z):
>             pass
>     
> >       l = getfuncargnames(f)
> 
> testing/python/integration.py:112:
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
> _pytest/compat.py:93: in getfuncargnames
>     startindex += num_mock_patch_args(function)
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
> 
> function = <function func at 0x7f00c1c779b0>
> 
>     def num_mock_patch_args(function):
>         """ return number of arguments used up by mock arguments (if any) """
>         patchings = getattr(function, "patchings", None)
>         if not patchings:
>             return 0
>         mock = sys.modules.get("mock", sys.modules.get("unittest.mock", None))
>         if mock is not None:
>             return len([p for p in patchings
> >                           if not p.attribute_name and p.new is mock.DEFAULT])
> E           AttributeError: 'str' object has no attribute 'attribute_name'
> 
> _pytest/compat.py:80: AttributeError
> =========================================================== 1 failed, 1683 passed, 45 skipped, 13 xfailed in 249.19 seconds ============================================================
> 
```

this happens with

```
> platform linux2 -- Python 2.7.13, pytest-3.0.7.dev0, py-1.4.31, pluggy-0.4.0
> rootdir: /home/marv/scm/github/pytest, inifile: tox.ini
> plugins: expect-1.1.0, hypothesis-3.6.0
```

as well as with pytest-3.0.5 and 3.0.6

**Comment by nicoddemus** [comment]
Hi,

The [recommended way](https://github.com/pytest-dev/pytest/blob/master/CONTRIBUTING.rst#id12) to test pytest is using `tox`, did you try that?

**Comment by nicoddemus** [comment]
Close from lack of response

**Comment by sbraz** [comment]
For future reference, this was fixed in 321f66f71148c978c1bf45dace61886b5e263bd4
